Sources/Core/MaterialTable.cpp

```diff
@@ -10,6 +12,21 @@ MaterialTable::~MaterialTable(void)
 {
 }
 
+void MaterialTable::UpdateMaterial(ShaderProgram *shaderProgram) noexcept
+{
+	auto materials = _shadersPrograms[shaderProgram];
+	
+	for (auto material: materials)
+	{
+		material->Update();
+	}
+}
+
+void	MaterialTable::AddToList(ShaderProgram * shaderProgram, Material * material)
```

Comment:
Is this function useful ?
I'd rather add the ShaderProgram when the material is added (inside the RegisterObject class)